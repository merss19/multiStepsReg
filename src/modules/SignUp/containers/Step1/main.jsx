"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var redux_form_1 = require("redux-form");
var react_router_dom_1 = require("react-router-dom");
var ducks_1 = require("./ducks");
var api_1 = require("./api");
var InputText_1 = require("components/InputText");
var RadioInput_1 = require("components/RadioInput");
var CheckboxInput_1 = require("components/CheckboxInput");
var Button_1 = require("components/Button");
var storage_1 = require("tools/storage");
var const_1 = require("../../const");
var CustomSteps_1 = require("../CustomSteps");
var CustomModal_1 = require("modules/CustomModal");
var ducks_2 = require("modules/Login/ducks");
var formValidations = require("tools/validations");
var ModalHOC_1 = require("components/HOC/ModalHOC");
var StepOneComponent = /** @class */ (function (_super) {
    __extends(StepOneComponent, _super);
    function StepOneComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StepOneComponent.prototype.shouldComponentUpdate = function (nextProps) {
        return (nextProps.step !== this.props.step);
    };
    StepOneComponent.prototype.onSubmit = function (data) {
        var _this = this;
        this.props.userCreate(data, function (data) { return _this.resultAction(data); });
    };
    StepOneComponent.prototype.resultAction = function (data) {
        var _a = this.props, changeStep = _a.changeStep, setToken = _a.setToken;
        if (data && data.authToken) {
            setToken(data.authToken);
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            storage_1.storage.save('promoName', data.promoName, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            changeStep(const_1.Steps.two);
        }
    };
    StepOneComponent.prototype.render = function () {
        var _this = this;
        var handleSubmit = this.props.handleSubmit;
        return (<div>
        {this.renderHeader()}
        <form className='entry__box' onSubmit={handleSubmit(function (data) { return _this.onSubmit(data); })}>
          <redux_form_1.Field name='email' placeholder='Ваш e-mail' type='text' component={InputText_1.default}/>
          <redux_form_1.Field name='password' placeholder='Ваш пароль' type='password' component={InputText_1.default}/>
          <redux_form_1.Field name='passwordAgain' placeholder='Пароль повторно' type='password' component={InputText_1.default}/>

          <div className='gender'>
            <div className='gender__group'>
              <p className='gender__title'>Ваш пол:</p>
              <redux_form_1.Field name='gender' id='male' value='male' type='radio' title='Мужчина' validate={formValidations.validateGender} component={RadioInput_1.default}/>
              <redux_form_1.Field name='gender' id='female' value='female' type='radio' title='Женщина' validate={formValidations.validateGender} component={RadioInput_1.default}/>
            </div>
            <hr className='gender__hr'/>
          </div>

          <div className='checkboxes'>
            <redux_form_1.Field name='accept' title='Принять условия ' id='accept' component={CheckboxInput_1.default}/>
          </div>

          <Button_1.Button type='submit' styleBtn={Button_1.ButtonTypes.info} wide={true}>
            Зарегистрироваться
          </Button_1.Button>


          <div className='entry__link text-center mtb20'>
            <div className='entry-nav__item'>
              <react_router_dom_1.Link to='/'>Войти</react_router_dom_1.Link>
            </div>
            <div className='entry-nav__item'>
              <react_router_dom_1.Link to='/restore'>Забыли пароль?</react_router_dom_1.Link>
            </div>
          </div>
        </form>
      </div>);
    };
    return StepOneComponent;
}(CustomSteps_1.default));
exports.StepOneComponent = StepOneComponent;
exports.StepOne = react_redux_1.connect(null, {
    toggleModal: CustomModal_1.toggleModal,
    setError: ducks_1.setError,
    userCreate: ducks_1.userCreate,
    setToken: ducks_2.setToken
})(StepOneComponent);
exports.default = ModalHOC_1.modalHOC(redux_form_1.reduxForm({
    form: 'StepOne',
    validate: formValidations.validateSignUp,
    asyncValidate: api_1.asyncValidate,
    asyncBlurFields: ['email']
})(exports.StepOne), ducks_1.moduleName, ducks_1.subModuleName);
