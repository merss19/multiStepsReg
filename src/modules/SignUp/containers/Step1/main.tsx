import * as React from 'react';
import {connect} from 'react-redux';
import {Field, FormProps, reduxForm} from 'redux-form';
import {Link} from 'react-router-dom';
import {setError, userCreate, moduleName, subModuleName} from './ducks';
import {asyncValidate} from './api';
import InputText from 'components/InputText';
import RadioInput from 'components/RadioInput';
import CheckboxInput from 'components/CheckboxInput';
import {SignUpFormData} from '../../interfaces';
import {Button, ButtonTypes} from 'components/Button';
import {storage} from 'tools/storage';
import {Steps} from '../../const';
import CustomSteps, {CustomStepsProps} from '../CustomSteps';
import {toggleModal} from "modules/CustomModal";
import {setToken} from 'modules/Login/ducks'
import * as formValidations from 'tools/validations'
import {modalHOC} from "components/HOC/ModalHOC";

interface OwnProps extends CustomStepsProps, FormProps<{}, {}, {}> {
  changeStep: (data: Steps) => void
  step: Steps
}

type ConnectedState = {
  isFetching: boolean
  isError: boolean
  errMsg: string
}
type ConnectedDispatch = {
  getPrograms(): void
  setToken(token: string): void
  toggleModal(data: boolean): void
  setError(data: boolean): any
  userCreate(data: SignUpFormData, cb: (data: any) => void): any
}

type Props = OwnProps & ConnectedDispatch;

export class StepOneComponent extends CustomSteps<Props> {

  shouldComponentUpdate(nextProps: Props) {
    return (
      nextProps.step !== this.props.step
    );
  }

  onSubmit(data: SignUpFormData) {
    this.props.userCreate(data, (data) => this.resultAction(data));
  }

  resultAction(data: any): void {
    const {changeStep, setToken} = this.props;
    if (data && data.authToken) {
      setToken(data.authToken);
      storage.save('token', data.authToken,
        {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      storage.save('promoName', data.promoName,
        {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      changeStep(Steps.two);
    }
  }

  render() {
    const {handleSubmit} = this.props;
    return (
      <div>
        {this.renderHeader()}
        <form className='entry__box' onSubmit={handleSubmit((data: any) => this.onSubmit(data))}>
          <Field
            name='email'
            placeholder='Ваш e-mail'
            type='text'
            component={InputText}
          />
          <Field
            name='password'
            placeholder='Ваш пароль'
            type='password'
            component={InputText}
          />
          <Field
            name='passwordAgain'
            placeholder='Пароль повторно'
            type='password'
            component={InputText}
          />

          <div className='gender'>
            <div className='gender__group'>
              <p className='gender__title'>Ваш пол:</p>
              <Field
                name='gender'
                id='male'
                value='male'
                type='radio'
                title='Мужчина'
                validate={formValidations.validateGender}
                component={RadioInput}
              />
              <Field
                name='gender'
                id='female'
                value='female'
                type='radio'
                title='Женщина'
                validate={formValidations.validateGender}
                component={RadioInput}
              />
            </div>
            <hr className='gender__hr'/>
          </div>

          <div className='checkboxes'>
            <Field
              name='accept'
              title='Принять условия '
              id='accept'
              component={CheckboxInput}/>
          </div>

          <Button
            type='submit'
            styleBtn={ButtonTypes.info}
            wide={true}
          >
            Зарегистрироваться
          </Button>


          <div className='entry__link text-center mtb20'>
            <div className='entry-nav__item'>
              <Link to='/'>Войти</Link>
            </div>
            <div className='entry-nav__item'>
              <Link to='/restore'>Забыли пароль?</Link>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export const StepOne = connect(null, {
  toggleModal,
  setError,
  userCreate,
  setToken
})(StepOneComponent);

export default modalHOC(reduxForm({
  form: 'StepOne',
  validate: formValidations.validateSignUp,
  asyncValidate,
  asyncBlurFields: ['email']
})(StepOne), moduleName, subModuleName);