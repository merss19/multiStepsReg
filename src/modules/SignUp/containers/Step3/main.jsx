"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var react_router_dom_1 = require("react-router-dom");
var CustomSteps_1 = require("../CustomSteps");
var const_1 = require("../../const");
var Button_1 = require("components/Button");
var Login_1 = require("modules/Login");
var FetchDucks_1 = require("modules/FetchDucks");
var ModalHOC_1 = require("components/HOC/ModalHOC");
var ducks_1 = require("./ducks");
var selectors_1 = require("../../selectors");
var _a = FetchDucks_1.fetchDucks(ducks_1.moduleName, ducks_1.subModuleName), selectIsFetching = _a.selectIsFetching, selectIsLoad = _a.selectIsLoad, selectIsError = _a.selectIsError, selectErrMsg = _a.selectErrMsg;
var mapStateToProps = function (state) { return ({
    authToken: Login_1.selectAuthToken(state),
    currentPackage: selectors_1.selectPackage(state),
    choosenProgram: selectors_1.selectChoosenProgram(state),
    payInfo: selectors_1.selectPayInfo(state)
}); };
var StepThreeComponent = /** @class */ (function (_super) {
    __extends(StepThreeComponent, _super);
    function StepThreeComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StepThreeComponent.prototype.componentWillMount = function () {
        var _a = this.props, authToken = _a.authToken, paymentInfo = _a.paymentInfo;
        paymentInfo(authToken);
    };
    StepThreeComponent.prototype.shouldComponentUpdate = function (nextProps) {
        return (nextProps.step !== this.props.step
            || nextProps.authToken !== this.props.authToken
            || nextProps.choosenProgram !== this.props.choosenProgram
            || nextProps.payInfo !== this.props.payInfo
            || nextProps.currentPackage !== this.props.currentPackage);
    };
    StepThreeComponent.prototype.payManual = function () {
        var _a = this.props, paymentManual = _a.paymentManual, payInfo = _a.payInfo, changeStep = _a.changeStep;
        paymentManual(payInfo.get('txId'), function (data) {
            changeStep(const_1.Steps.four);
        });
    };
    StepThreeComponent.prototype.back = function (e) {
        var changeStep = this.props.changeStep;
        changeStep(const_1.Steps.two);
    };
    StepThreeComponent.prototype.renderContent = function () {
        var _this = this;
        var _a = this.props, payInfo = _a.payInfo, changeStep = _a.changeStep;
        if (payInfo.get('amount') !== 0) {
            if (payInfo.get('isPaid')) {
                return (<div className='entry__box'>
            <div className='entry-ico-box'>
              <img className='entry-ico-box__img' src={'/img/ico-success.svg'} alt=""/>
              <p className='entry-ico-box__title'>Оплачен!</p>
            </div>

            <Button_1.Button type={'button'} styleBtn={Button_1.ButtonTypes.info} wide={true}>
              <react_router_dom_1.Link to="/profile">Продолжить</react_router_dom_1.Link>
            </Button_1.Button>
          </div>);
            }
            else {
                return (<div className='entry__box'>
            <h3 className='text-center mtb20'>Выберите способ оплаты:</h3>
            <ul className='options--with-img wide mb20'>
              <li className='options__item'>
                <form id='yaForm' ref={function (el) { return _this.yandex = el; }} action='https://money.yandex.ru/eshop.xml' className='form-pay' target='_blank' method='POST'>
                  <input name='shopId' defaultValue='' type='hidden'/>
                  <input name='scid' defaultValue='' type='hidden'/>
                  <input name='customerNumber' defaultValue={'HertysKJ'} type='hidden'/>
                  <input name='sum' defaultValue={payInfo.get('amount')} type='hidden'/>
                  <input name='orderNumber' defaultValue={payInfo.get('txId')} type='hidden'/>
                  <input name='paymentType' defaultValue='' type='hidden'/>
                  <div className='form-pay__submit' onClick={function () {
                    _this.yandex.submit();
                }}>
                    <img className='ico-yandexkassa mt5' src={process.env.PUBLIC_URL + '/img/png/pay/yandexkassa.png'} alt=''/>
                  </div>

                </form>
              </li>
              <li className='options__item'>
                <form id='robokassaForm' action='https://auth.robokassa.ru/Merchant/Index.aspx' ref={function (el) { return _this.robokassa = el; }} className='form-pay' target='_blank' method='POST'>
                  <input type='hidden' name='MrchLogin' defaultValue=''/>
                  <input type='hidden' name='OutSum' defaultValue={payInfo.get('amount')}/>
                  <input type='hidden' name='Desc' defaultValue='Text description'/>
                  <input type='hidden' name='SignatureValue' defaultValue=''/>
                  <input type='hidden' name='shp_txid' defaultValue={payInfo.get('txId')}/>
                  <input type='hidden' name='Culture' defaultValue='ru'/>
                  <div className='form-pay__submit' onClick={function () {
                    _this.robokassa.submit();
                }}>
                    <img className='ico-robokassa mt20' src={'/img/png/pay/robokassa.png'} alt=''/>
                  </div>

                </form>
              </li>
            </ul>

            <ul className='entry-nav'>
              <li className='entry-nav__item'>
                <a href='#' className='entry-nav__link' onClick={function (e) {
                    _this.back(e);
                }}>Вернуться на шаг назад</a>
              </li>
            </ul>

            <ul className='entry-nav'>
              <li className='entry-nav__item'>
                <react_router_dom_1.Link to='/' onClick={function () { return _this.logOut(); }}>Выйти</react_router_dom_1.Link>
              </li>
            </ul>
          </div>);
            }
        }
        else {
            return (<div className='entry__box'>
          <div className='entry-ico-box '>
            <img className='entry-ico-box__img' src='/assets/img/svg/ico-freebie.svg' alt=''/>
            <p className='entry-ico-box__title'>Халява!</p>
          </div>
          <Button_1.Button onClick={function () { return _this.payManual(); }} type={'button'} styleBtn={Button_1.ButtonTypes.info} wide={true}>
            Продолжить
          </Button_1.Button>
        </div>);
        }
    };
    StepThreeComponent.prototype.renderPrice = function () {
        var payInfo = this.props.payInfo;
        if (!payInfo.get('isPaid')) {
            return (<p className='entry-program-price highlight md30'>
          {payInfo.get('amount')} руб.
        </p>);
        }
        return null;
    };
    StepThreeComponent.prototype.render = function () {
        var _a = this.props, choosenProgram = _a.choosenProgram, currentPackage = _a.currentPackage;
        return (<div>
        {this.renderHeader()}
        {choosenProgram % const_1.ProgramNum.count !== 0 && <p className='entry-packet-name'>{currentPackage.get('name')}</p>}
        {this.renderPrice()}
        {this.renderContent()}
      </div>);
    };
    return StepThreeComponent;
}(CustomSteps_1.default));
exports.StepThreeComponent = StepThreeComponent;
exports.StepThree = react_redux_1.connect(mapStateToProps, {
    paymentInfo: ducks_1.paymentInfo,
    paymentManual: ducks_1.paymentManual
})(StepThreeComponent);
exports.default = ModalHOC_1.modalHOC(exports.StepThree, ducks_1.moduleName, ducks_1.subModuleName);
