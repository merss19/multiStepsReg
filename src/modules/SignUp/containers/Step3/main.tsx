import * as React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import CustomSteps, {CustomStepsProps} from '../CustomSteps';
import {Steps, ProgramNum} from '../../const'
import {Button, ButtonTypes} from 'components/Button'
import * as shortid from 'shortid'
import * as md5 from 'md5'
import {selectAuthToken} from 'modules/Login';
import {fetchDucks} from 'modules/FetchDucks';
import {modalHOC} from 'components/HOC/ModalHOC';
import {
  paymentInfo,
  paymentManual,
  moduleName,
  subModuleName
} from './ducks';

import {
  selectChoosenProgram,
  selectPackage,
  selectPayInfo
} from '../../selectors';

interface OwnProps extends CustomStepsProps {
  changeStep: (data: number) => void
  step: Steps
}
const {
  selectIsFetching,
  selectIsLoad,
  selectIsError,
  selectErrMsg
} = fetchDucks(moduleName, subModuleName);

export type ConnectedState = {
  authToken: string
  choosenProgram: number
  currentPackage: any
  payInfo: any
}

type ConnectedDispatch = {
  paymentInfo(data: string): void
  paymentManual(data: string, cb: (data: boolean) => void): void
}

const mapStateToProps = (state: any): ConnectedState => ({
  authToken: selectAuthToken(state),
  currentPackage: selectPackage(state),
  choosenProgram: selectChoosenProgram(state),
  payInfo: selectPayInfo(state)
});
type Props = OwnProps & ConnectedDispatch & ConnectedState

export class StepThreeComponent extends CustomSteps<Props> {

  yandex: any;
  robokassa: any;

  componentWillMount() {
    const {authToken, paymentInfo} = this.props;
    paymentInfo(authToken);
  }
  shouldComponentUpdate(nextProps:Props){
    return (
      nextProps.step !== this.props.step
      ||  nextProps.authToken !== this.props.authToken
      ||  nextProps.choosenProgram !== this.props.choosenProgram
      ||  nextProps.payInfo !== this.props.payInfo
      ||  nextProps.currentPackage !== this.props.currentPackage
    );
  }

  payManual() {
    const {paymentManual, payInfo, changeStep} = this.props;
    paymentManual(payInfo.get('txId'), (data) => {
      changeStep(Steps.four);
    })
  }

  back(e: any) {
    const {changeStep} = this.props;
    changeStep(Steps.two);
  }

  renderContent() {
    const {payInfo, changeStep} = this.props;
    if (payInfo.get('amount') !== 0) {
      if (payInfo.get('isPaid')) {
        return (
          <div className='entry__box'>
            <div className='entry-ico-box'>
              <img className='entry-ico-box__img' src={'/img/ico-success.svg'} alt=""/>
              <p className='entry-ico-box__title'>Оплачен!</p>
            </div>

            <Button
              type={'button'}
              styleBtn={ButtonTypes.info}
              wide={true}
            >
              <Link to="/profile">Продолжить</Link>
            </Button>
          </div>
        );
      } else {
        return (
          <div className='entry__box'>
            <h3 className='text-center mtb20'>Выберите способ оплаты:</h3>
            <ul className='options--with-img wide mb20'>
              <li className='options__item'>
                <form
                  id='yaForm'
                  ref={(el) => this.yandex = el}
                  action='https://money.yandex.ru/eshop.xml'
                  className='form-pay'
                  target='_blank'
                  method='POST'
                >
                  <input name='shopId' defaultValue='' type='hidden'/>
                  <input name='scid' defaultValue='' type='hidden'/>
                  <input name='customerNumber' defaultValue={'HertysKJ'} type='hidden'/>
                  <input name='sum' defaultValue={payInfo.get('amount')} type='hidden'/>
                  <input name='orderNumber' defaultValue={payInfo.get('txId')} type='hidden'/>
                  <input name='paymentType' defaultValue='' type='hidden'/>
                  <div
                    className='form-pay__submit'
                    onClick={() => {
                      this.yandex.submit()
                    }}
                  >
                    <img
                      className='ico-yandexkassa mt5'
                      src={process.env.PUBLIC_URL + '/img/png/pay/yandexkassa.png'}
                      alt=''
                    />
                  </div>

                </form>
              </li>
              <li className='options__item'>
                <form
                  id='robokassaForm'
                  action='https://auth.robokassa.ru/Merchant/Index.aspx'
                  ref={(el) => this.robokassa = el}
                  className='form-pay'
                  target='_blank'
                  method='POST'
                >
                  <input type='hidden' name='MrchLogin' defaultValue=''/>
                  <input type='hidden' name='OutSum' defaultValue={payInfo.get('amount')}/>
                  <input type='hidden' name='Desc' defaultValue='Text description'/>
                  <input
                    type='hidden'
                    name='SignatureValue'
                    defaultValue=''
                  />
                  <input type='hidden' name='shp_txid' defaultValue={payInfo.get('txId')}/>
                  <input type='hidden' name='Culture' defaultValue='ru'/>
                  <div
                    className='form-pay__submit'
                    onClick={() => {
                      this.robokassa.submit()
                    }}
                  >
                    <img
                      className='ico-robokassa mt20'
                      src={'/img/png/pay/robokassa.png'} alt=''
                    />
                  </div>

                </form>
              </li>
            </ul>

            <ul className='entry-nav'>
              <li className='entry-nav__item'>
                <a href='#' className='entry-nav__link' onClick={e => {
                  this.back(e)
                }}>Вернуться на шаг назад</a>
              </li>
            </ul>

            <ul className='entry-nav'>
              <li className='entry-nav__item'>
                <Link to='/' onClick={() => this.logOut()}>Выйти</Link>
              </li>
            </ul>
          </div>
        );
      }
    } else {
      return (
        <div className='entry__box'>
          <div className='entry-ico-box '>
            <img
              className='entry-ico-box__img'
              src='/assets/img/svg/ico-freebie.svg' alt=''
            />
            <p className='entry-ico-box__title'>Халява!</p>
          </div>
          <Button
            onClick={() => this.payManual()}
            type={'button'}
            styleBtn={ButtonTypes.info}
            wide={true}
          >
            Продолжить
          </Button>
        </div>
      );
    }

  }

  renderPrice() {
    const {payInfo} = this.props;
    if (!payInfo.get('isPaid')) {
      return (
        <p className='entry-program-price highlight md30'>
          {payInfo.get('amount')} руб.
        </p>
      );
    }
    return null;
  }

  render() {
    const {
      choosenProgram,
      currentPackage,
    } = this.props;
    return (
      <div>
        {this.renderHeader()}
        {choosenProgram % ProgramNum.count !== 0 && <p className='entry-packet-name'>{currentPackage.get('name')}</p>}
        {this.renderPrice()}
        {this.renderContent()}
      </div>
    );
  }
}

export const StepThree = connect(mapStateToProps, {
  paymentInfo,
  paymentManual
})(StepThreeComponent);
export default modalHOC(StepThree, moduleName, subModuleName);
