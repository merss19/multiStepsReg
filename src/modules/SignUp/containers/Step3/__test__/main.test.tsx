import {shallow, mount} from 'enzyme';
import {StepThreeComponent, StepThree} from '../main';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import * as sinon from 'sinon';
import {Provider} from 'react-redux';
import configureStore from '../../../../../store';
import {BrowserRouter as Router, Link} from 'react-router-dom';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {storage} from '../../../../../tools/storage/index';
import {Steps} from '../../../const';
import {Button, ButtonTypes} from "../../../../../components/Button/button";
import * as Immutable from 'immutable';

const store = configureStore();
configure({adapter: new Adapter()});
let data = [
  {
    id: 1,
    name: 'hero',
    isActive: true
  },
  {
    id: 2,
    name: 'mother',
    isActive: false
  },
  {
    id: 3,
    name: 'extreme',
    isActive: false
  },
  {
    id: 4,
    name: 'tomorrow',
    isActive: false
  }
];

let changeStep,
  handleSubmit,
  setToken,
  toggleModal,
  setPackageType,
  authToken,
  payInfo,
  paymentInfo,
  paymentManual,
  setError,
  isFetching,
  isLoad,
  isError,
  step,
  errMsg,
  programs,
  choosenPackageType,
  choosenProgram,
  packages,
  currentPackage,
  promo,
  promoError,
  match,
  location,
  callback,
  history;
const start = () => {
  beforeEach(() => {
    changeStep = jest.fn(),
    callback = jest.fn(),
    paymentManual = jest.fn((payload, callback) => callback())
    setToken = jest.fn(),
    handleSubmit = jest.fn((callback) => callback()),
    toggleModal = jest.fn(),
    setError = jest.fn(),
    paymentInfo = jest.fn(),
    authToken = 'test',
    payInfo = Immutable.fromJS({
      txId: 'qweqwe',
      isPaid: true,
      amount: 3000
    }),
    isFetching = false,
    isLoad = false,
    isError = false,
    errMsg = 'error',
    step = Steps.one,
    packages = Immutable.fromJS([
        {
          name: '1 человек',
          isActive: false,
          id: 1
        },
        {
          name: '2 человек',
          isActive: false,
          id: 2
        },
        {
          name: '1 ЧЕЛОВЕК + ФИТНЕС-БРАСЛЕТ',
          isActive: false,
          id: 3
        },
        {
          name: '1 ЧЕЛОВЕК + КОВРИК',
          isActive: false,
          id: 4
        },
        {
          name: '3 человек',
          isActive: false,
          id: 5
        },
      ]) ,
      currentPackage = Immutable.fromJS({
        cost: 3000,
        name: '1 человек',
        isActive: true,
        id: 1
      }),
      promo = '',
      promoError = '',
      programs = Immutable.fromJS(data),
      /*signUpData = {
        email: 'mail@mail.ru',
        password: 'qweqwe',
        passwordAgain: 'qweqwe',
        gender: 'male',
      },*/
      match = {
        isExact: true,
        params: {},
        path: "/task",
        url: "/task"
      },
      location = {},
      history = {
        replace: jest.fn()
      }
  });
  const props = {
    handleSubmit: handleSubmit,
    changeStep: changeStep,
    paymentInfo: paymentInfo,
    paymentManual: paymentManual,
    toggleModal: toggleModal,
    setError: setError,
    authToken: authToken,
    payInfo: payInfo,
    setPackageType: setPackageType,
    step: step,
    choosenPackageType: choosenPackageType,
    choosenProgram: choosenProgram,
    currentPackage: currentPackage,
    errMsg: errMsg,
    isFetching: isFetching,
    isLoad: isLoad,
    isError: isError,
    match: match,
    location: location,
    history: history
  };
  const buildComponent = () => {
    return shallow(<StepThreeComponent {...props}/>)
  };
  const buildComponentMount = () => {
    return mount(
      <Provider store={store}>
        <Router>
          <StepThree {...props}/>
        </Router>
      </Provider>)
  };
  return {
    buildComponentMount,
    props,
    buildComponent
  }
};
describe('Component StepThree', () => {
  let component;
  start();
  beforeEach(() => {
    component = start().buildComponent();
  });
  it('renders component', () => {
    expect(component).toHaveLength(1);
  });
  it('renders component Mount', () => {
    const component = start().buildComponentMount();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('call paymentInfo', () => {
    component = start().buildComponent();
    expect(paymentInfo).toHaveBeenCalled()
  });
  it('call payManual', () => {
    const component = start().buildComponent();
    const instance = component.instance() as StepThreeComponent;
    instance.payManual();
    expect(paymentManual).toHaveBeenCalled();
    //expect(paymentCreate).toHaveBeenCalledWith(payload, cb);
  });
  it('call back', () => {
    const component = start().buildComponent();
    const instance = component.instance() as StepThreeComponent;
    const e = {
      preventDefault: () => true
    }
    instance.back(e);
    expect(changeStep).toHaveBeenCalledWith(Steps.two);
  });
  describe('Call renderContent', () => {
    it('amount !== 0, isPaid === true', () => {
      const component = start().buildComponent();
      expect(component.find('.entry-ico-box__title').text()).toBe('Оплачен!')
      expect(component.find(Button).exists()).toBe(true)
    });
    it('amount !== 0, isPaid === false', () => {
      payInfo = Immutable.fromJS({
        txId: 'qweqwe',
        isPaid: false,
        amount: 3000
      });
      const component = start().buildComponent();
      expect(component.find('.form-pay')).toHaveLength(2);
      expect(component.find('.entry-nav')).toHaveLength(2);
      expect(component.find(Button).exists()).toBe(false)
    });
    it('amount === 0', () => {
      payInfo = Immutable.fromJS({
        txId: 'qweqwe',
        isPaid: false,
        amount: 0
      });
      const component = start().buildComponent();
      console.log('step333')
      console.log(component.find('.entry-ico-box__img').prop('src'))
      expect(component.find('.entry-ico-box__title').text()).toBe('Халява!')
      //expect(component.find('.entry-nav')).toHaveLength(2);
      expect(component.find('.entry-ico-box__title').text()).toBe('Халява!')
      expect(component.find(Button).exists()).toBe(true)
      expect(component.find('.entry-ico-box__img').prop('src')).toBe('/assets/img/svg/ico-freebie.svg')
    });
  });
  it('call renderPrice, isPaid === true', () => {
    const component = start().buildComponent();
    const instance = component.instance() as StepThreeComponent;
    instance.renderPrice();
    expect(component.find('.entry-program-price')).toHaveLength(0);
  });
  it('call renderPrice, isPaid === false', () => {
    payInfo = Immutable.fromJS({
      txId: 'qweqwe',
      isPaid: false,
      amount: 3000
    });
    const component = start().buildComponent();
    const instance = component.instance() as StepThreeComponent;
    instance.renderPrice();
    expect(component.find('.entry-program-price')).toHaveLength(1);
    expect(component.find('.entry-program-price').text()).toBe(start().props.payInfo.get('amount') + ' руб.');
  });
  it('click back', () => {
    payInfo = Immutable.fromJS({
      txId: 'qweqwe',
      isPaid: false,
      amount: 3000
    });
    const component = start().buildComponent();
    const spy = sinon.spy(StepThreeComponent.prototype, 'back');
    const button = component.find('.entry-nav').at(0).find('.entry-nav__link');
    button.simulate('click');
    expect(spy.called).toBe(true);
  });
  it('click logOut', () => {
    payInfo = Immutable.fromJS({
      txId: 'qweqwe',
      isPaid: false,
      amount: 3000
    });
    const component = start().buildComponent();
    const spy = sinon.spy(StepThreeComponent.prototype, 'logOut');
    const button = component.find('.entry-nav').at(1).find(Link);
    button.simulate('click');
    expect(spy.called).toBe(true);
  });
  it('click payManual', () => {
    payInfo = Immutable.fromJS({
      txId: 'qweqwe',
      isPaid: false,
      amount: 0
    });
    const component = start().buildComponent();
    const spy = sinon.spy(StepThreeComponent.prototype, 'payManual');
    const button = component.find(Button);
    button.simulate('click');
    expect(spy.called).toBe(true);
  });
  describe('Test shouldComponentUpdate', () => {
    it('step', () => {
      const instance = component.instance() as StepThreeComponent;
      step = Steps.three;
      let props = {
        ...start().props,
        step: step
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('authToken', () => {
      const instance = component.instance() as StepThreeComponent;
      authToken = 'qwerty';
      let props = {
        ...start().props,
        authToken: authToken
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('payInfo', () => {
      const instance = component.instance() as StepThreeComponent;
      payInfo = {
        txId: 'qweqwe',
        test: {}
      };
      let props = {
        ...start().props,
        payInfo: payInfo
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('choosenProgram', () => {
      const instance = component.instance() as StepThreeComponent;
      choosenProgram = 3;
      let props = {
        ...start().props,
        choosenProgram: choosenProgram
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('currentPackage', () => {
      const instance = component.instance() as StepThreeComponent;
      currentPackage = Immutable.fromJS({
        name: '2 человек',
        isActive: true,
        id: 2
      });
      let props = {
        ...start().props,
        currentPackage: currentPackage
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
  });
});


