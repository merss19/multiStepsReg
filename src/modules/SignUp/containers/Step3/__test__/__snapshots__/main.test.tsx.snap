// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component StepThree renders correctly 1`] = `
<div>
  <div
    className="entry__header"
  >
    <h2
      className="entry__title entry__title--auth text-center "
    >
      Регистрация
    </h2>
  </div>
  <p
    className="entry-packet-name"
  >
    1 человек
  </p>
  <div
    className="entry__box"
  >
    <div
      className="entry-ico-box"
    >
      <img
        alt=""
        className="entry-ico-box__img"
        src="/img/ico-success.svg"
      />
      <p
        className="entry-ico-box__title"
      >
        Оплачен!
      </p>
    </div>
    <Component
      prefix="btn"
      styleBtn="secondary"
      type="button"
      wide={true}
    >
      <Link
        replace={false}
        to="/profile"
      >
        Продолжить
      </Link>
    </Component>
  </div>
</div>
`;
